### PR TITLE
Move the span operations inside the scope in the http server example

### PR DESCRIPTION
**Pull request by jkwatson**
This shows better idiomatic usage of the API.

### CODE REVIEW

File: examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java [hunk 1]
```diff
@@ -55,25 +55,26 @@ public void handle(HttpExchange he) throws IOException {
...
+         - http.scheme, http.host, http.target
+         - http.scheme, http.server_name, net.host.port, http.target
+         - http.scheme, net.host.name, net.host.port, http.target
+         - http.url
+        */
+        span.setAttribute("http.scheme", "http");
+        span.setAttribute("http.host", "localhost:" + HttpServer.port);
+        span.setAttribute("http.target", "/");
+        // Process the request
+        answer(he, span);
```
Comment (carlosalberto) [inline examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java:73]: (This `heh` golang style param name distracts me, hah)

Comment (jkwatson) [reply examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java]: fixed!

File: examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java [hunk 2]
```diff
@@ -40,40 +40,41 @@
...
+        */
+        span.setAttribute("http.scheme", "http");
+        span.setAttribute("http.host", "localhost:" + HttpServer.port);
+        span.setAttribute("http.target", "/");
+        // Process the request
+        answer(exchange, span);
+      } finally {
+        // Close the span
+        span.end();
+      }
```
Comment (bogdandrutu) [inline examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java:77]: Confused a bit and I would rewrite:
```
      // Extract the context from the HTTP request
      Context context =
          OpenTelemetry.getPropagators().getHttpTextFormat().extract(Context.current(), he, getter);
          OpenTelemetry.getPropagators().getHttpTextFormat().extract(Context.current(), exchange, getter);

      Span.Builder spanBuilder = tracer.spanBuilder("/").setSpanKind(Span.Kind.SERVER).setParent(TraceContextUtils.getSpan(context));
      Span span = spanBuilder.startSpan();

try (Scope scope = TraceContextUtils.withSpan(span)) {
        // Set the Semantic Convention
        span.setAttribute("component", "http");
        span.setAttribute("http.method", "GET");
        /*
         One of the following is required:
         - http.scheme, http.host, http.target
         - http.scheme, http.server_name, net.host.port, http.target
         - http.scheme, net.host.name, net.host.port, http.target
         - http.url
        */
        span.setAttribute("http.scheme", "http");
        span.setAttribute("http.host", "localhost:" + HttpServer.port);
        span.setAttribute("http.target", "/");
        // Process the request
        answer(exchange, span);
      } finally {
        // Close the span
        span.end();
      }
```

Comment (jkwatson) [reply examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java:77]: good point. I will update.

Comment (jkwatson) [reply examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java]: done!

File: examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java [hunk 3]
```diff
@@ -40,40 +40,41 @@
...
       Span.Builder spanBuilder = tracer.spanBuilder("/").setSpanKind(Span.Kind.SERVER);
-      Span span = null;
 
       // Extract the context from the HTTP request
       Context ctx =
-          OpenTelemetry.getPropagators().getHttpTextFormat().extract(Context.current(), he, getter);
+          OpenTelemetry.getPropagators().getHttpTextFormat().extract(Context.current(), exchange, getter);
+
+      Span span = spanBuilder.startSpan();
       try (Scope scope = ContextUtils.withScopedContext(ctx)) {
```
Comment (bogdandrutu) [inline examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java:56]: This should be with the new Span.

File: examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java [hunk 4]
```diff
@@ -40,40 +40,41 @@
...
+    public void handle(HttpExchange exchange) throws IOException {
       // Name convention for the Span is not yet defined.
       // See: https://github.com/open-telemetry/opentelemetry-specification/issues/270
       Span.Builder spanBuilder = tracer.spanBuilder("/").setSpanKind(Span.Kind.SERVER);
-      Span span = null;
 
       // Extract the context from the HTTP request
       Context ctx =
-          OpenTelemetry.getPropagators().getHttpTextFormat().extract(Context.current(), he, getter);
+          OpenTelemetry.getPropagators().getHttpTextFormat().extract(Context.current(), exchange, getter);
```
Comment (bogdandrutu) [inline examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java:53]: Need to set as parent the Span from the returned context in the Builder.

File: examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java (hunk 3 above)
Comment (jkwatson) [reply examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java:56]: you are correct. this example had some issues.  :)

File: examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java [hunk 5]
```diff
@@ -40,40 +40,46 @@
...
-        span = spanBuilder.startSpan();
+      Context context =
+          OpenTelemetry.getPropagators()
+              .getHttpTextFormat()
+              .extract(Context.current(), exchange, getter);
+
+      Span span =
+          tracer
+              .spanBuilder("/")
+              .setParent(TracingContextUtils.getSpan(context))
```
Comment (jkwatson) [inline examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java:59]: is this the best way to get the parent?

Comment (carlosalberto) [reply examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java:59]: (Makes me remember I've been longing to remove the dust from some code locally, to have a `setContext(Context context)` overload).

File: examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java (hunk 3 above)
Comment (carlosalberto) [reply examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java]: Probably that was me. I updated this example in a hurry among all the Context Prop changes and may have missed important bits here and there ;(

**Pull request merged**